raigad/src/main/java/com/netflix/raigad/dataobjects/NodeHealthChecker.java

```diff
@@ -0,0 +1,82 @@
+package com.netflix.raigad.dataobjects;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.netflix.raigad.configuration.IConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Created by alfasi on 4/15/15.
+ *
+ * NodeHealthChecker verifies that Elasticsearch process is running on the instance
+ * In case ES is up it should return the status 200 (Integer), otherwise it returns 500
+ */
+@Singleton
+public class NodeHealthChecker {
+
+    private static final Logger logger = LoggerFactory.getLogger(NodeHealthChecker.class);
+    private final IConfiguration config;
+
+    @Inject
+    public NodeHealthChecker(IConfiguration config) {
+        this.config = config;
+    }
+
+    public int isEsUpOnInstance() {
+        return parseResponse(getStatusWithRetries(1));
+    }
+
+    protected int parseResponse(String response) {
```

Comment:
If Json needs to be parsed, please use JsonCreator which creates Java Object from Json Response. 
Please look at the "objectmapper" and "dataobjects" packages for reference.
This also helps us to keep all Json parsing at a centralized place.